modules/con_notifications-configuration.adoc

```diff
@@ -0,0 +1,100 @@
+:_mod-docs-content-type: CONCEPT
+[id="notifications-configuration_{context}"]
+= NotificationsConfiguration
+
+The `NotificationsConfiguration` is a Kubernetes custom resource (CR) that manages notifications in a Kubernetes cluster. In {gitops-title}, the `NotificationsConfiguration` (CR) allows users to add Templates, Triggers, Services, and Subscriptions resources to an Argo CD Notifications ConfigMap.
+
+When you create a cluster in {gitops-title} with notifications enabled, a `NotificationsConfiguration` (CR) is created by default with the name `default-notifications-configuration`.
+
+Any change made in the existing configuration of the `NotificationsConfiguration` is replicated in Argo CD Notifications ConfigMap. For example, if the user adds some trigger configuration in the `NotificationsConfiguration` resource, this configuration is read, processed, and updated in the Argo CD Notifications ConfigMap.
+
+[NOTE]
+====
+- Any configuration changes must be updated in `default-notifications-configuration`. Custom resources created by the users for `NotificationsConfiguration` are not supported.
+- Any modification to the Argo CD Notifications ConfigMap, `argocd-notifications-cm`, are overriden by the changes made in the `NotificationsConfiguration`.
+====
+
+The `NotificationsConfiguration` Custom Resource consists of the following properties.
+
+* `Templates` - Templates are used to generate the notification template message.
+* `Triggers` - Triggers are used to define the condition when a notification is sent to the user and the list of templates required to generate the message.
+* `Services` - Services are used to deliver message.
+* `Subscriptions` - Subscriptions contain centrally managed global application subscriptions.
+
+The default value for all these resources is `empty`.
+
+The following examples define how to add Templates, Triggers, Services and Subscriptions resources to an Argo CD Notifications ConfigMap, `argocd-notification-cm`, using the `default-notifications-configuration` custom resource.
+
+.An example for `Templates`
+[source,yaml]
+----
+yaml
+apiVersion: argoproj.io/v1alpha1
+kind: NotificationsConfiguration
+metadata:
+ name: default-notifications-configuration
+spec:
+ templates:
+  template.my-custom-template: |
+    message: |
+     Application details: {{.context.argocdUrl}}/applications/{{.app.metadata.name}}.
+----
```

Comment:
Adding some explanations in the form of callouts to all the examples would be better. See: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#using-asciidoc-callouts-in-the-yaml
For example:
```suggestion
.An example for `Templates`
[source,yaml]
----
yaml
apiVersion: argoproj.io/v1alpha1
kind: NotificationsConfiguration
metadata:
 name: default-notifications-configuration
spec:
 templates: # <1>
  template.my-custom-template: | 
    message: | # <2>
     Application details: {{.context.argocdUrl}}/applications/{{.app.metadata.name}}.
----
<1> Explanation 1.
<2> Explanation 2.
```

Reply:
Fixed.